src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -2529,6 +2531,28 @@ public virtual void AttachTo(string entitySetName, object entity, string etag)
             this.entityTracker.InternalAttachEntityDescriptor(descriptor, true /*failIfDuplicated*/);
         }
 
+        /// <summary>Changes the state of the specified object to be deleted in the <see cref="Microsoft.OData.Client.DataServiceContext" />.</summary>
+        /// <param name="entity">The tracked entity to be changed to the deleted state.</param>
+        /// <param name="dependsOnIds">DependsOnIds for the deleted entity.</param>
+        /// <exception cref="System.ArgumentNullException">When <paramref name="entity" /> is null.</exception>
+        /// <exception cref="System.InvalidOperationException">When the object is not being tracked by the <see cref="Microsoft.OData.Client.DataServiceContext" />.</exception>
+        /// <remarks>
+        /// Existing objects in the Added state become detached.
+        /// </remarks>
+        public virtual void DeleteObject(object entity, params object[] dependsOnIds)
+        {
+            var dependsOn = new List<string>();
+            if (dependsOnIds != null)
+            {
+                foreach (var id in dependsOnIds)
+                {
+                    dependsOn.Add(id.ToString());

```

Comment:
This is repeated in the file in `UpdateObject` refactor to reduce repetition.

Reply:
Done!